src/libs/ReportUtils.ts

```diff
@@ -90,7 +90,7 @@ type ReportAndWorkspaceName = {
 
 type OptimisticReportAction = {
     commentText: string;
-    reportAction: Partial<ReportAction>;
+    reportAction: {reportActionID: string} & Partial<ReportAction>;
```

Comment:
What is the reason for this?

Reply:
reportActionID will be always there so there is no need to make it optional that why this change was made

Reply:
I dont know if its a good approach because in this case to build `optimisticAddCommentReportAction` its used `buildOptimisticTaskCommentReportAction` function which inside is calling which inside is calling `buildOptimisticAddCommentReportAction` where this
```
type OptimisticReportAction = {
    commentText: string;
    reportAction: {reportActionID: string} & Partial<ReportAction>;
};
``` 
type is used , so basically speaking everywhere where we are building `optimistic{something}ReportAction` we are using type 

```
type OptimisticReportAction = {
    commentText: string;
    reportAction: {reportActionID: string} & Partial<ReportAction>;
};
```

becuase here its how the function looks like for this 
```

function buildOptimisticAddCommentReportAction(text?: string, file?: File): OptimisticReportAction {
    const parser = new ExpensiMark();
    const commentText = getParsedComment(text ?? '');
    const isAttachment = !text && file !== undefined;
    const attachmentInfo = isAttachment ? file : {};
    const htmlForNewComment = isAttachment ? CONST.ATTACHMENT_UPLOADING_MESSAGE_HTML : commentText;

    // Remove HTML from text when applying optimistic offline comment
    const textForNewComment = isAttachment ? CONST.ATTACHMENT_MESSAGE_TEXT : parser.htmlToText(htmlForNewComment);
    return {
        commentText,
        reportAction: {
            reportActionID: NumberUtils.rand64(),
            actionName: CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT,
            actorAccountID: currentUserAccountID,
            person: [
                {
                    style: 'strong',
                    text: allPersonalDetails?.[currentUserAccountID ?? -1]?.displayName ?? currentUserEmail,
                    type: 'TEXT',
                },
            ],
            automatic: false,
            avatar: allPersonalDetails?.[currentUserAccountID ?? -1]?.avatar ?? UserUtils.getDefaultAvatarURL(currentUserAccountID),
            created: DateUtils.getDBTime(),
            message: [
                {
                    translationKey: isAttachment ? CONST.TRANSLATION_KEYS.ATTACHMENT : '',
                    type: CONST.REPORT.MESSAGE.TYPE.COMMENT,
                    html: htmlForNewComment,
                    text: textForNewComment,
                },
            ],
            isFirstItem: false,
            isAttachment,
            attachmentInfo,
            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
            shouldShow: true,
        },
    };
}
```

maybe the name of the type could be changed but I think if its used in multiple places for building optimistic report action I would leave it as it is
let me know what do you think